Fix links and NeoForge mentions in README.md

Mishura4 opened the pull request:
(Totally not an excuse to snatch PR #1)

Mishura4 committed:
Fix links and NeoForge mentions in README.md

embeddedt commented on docs/README.md:
```diff
@@ -5,7 +5,7 @@ NeoForge
 
 
-Forge is a free, open-source modding API all of your favourite mods use!
+NeoForge is a free, open-source modding API all of your favourite mods use!
```
Third time's the charm?

```suggestion
NeoForge is a free, open-source modding API many of your favourite mods use!
```

marchermans committed:
Update docs/README.md

Co-authored-by: embeddedt <42941056+embeddedt@users.noreply.github.com>

cpw commented:
We'll have to see if this still works after the big update that is currently sitting on my computer, heh.

SizableShrimp reviewed (changes requested):
I went ahead and rebased this PR on the newest 1.20.x changed git history for the benefit of the PR author.
Now for my thoughts:
If we're updating the README, might as well do it right.

> Installing NeoForged

NeoForged is the organization, not the product.
Additionally, this section is bad in general. It links to a dead placeholder curseforge link, which can be confusing to users. And, the information about Latest builds being more unstable is basically bunk nowadays. We should probably delete this section while we figure out what our installation will look like.

> NeoForged is a large project with many collaborators working on it around the clock.

Same issue, it's NeoForge, not NeoForged.

The CLA link and CONTRIBUTING.md doc are both outdated, but those should probably be saved for changing on another day.

hypersmc commented:
Please update the download button from the link: https://maven.neoforged.net/releases/net/neoforged/forge/
to: https://maven.neoforged.net/#/releases/net/neoforged/forge/

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/neoforged/NeoForge?pullRequest=1) <br/>All committers have signed the CLA.

cpw commented:
Updated to latest tip so this should run all the checks properly and then be ready to merge.

Mishura4 committed:
Update docs/README.md

See conversation at https://github.com/neoforged/NeoForge/pull/1#pullrequestreview-1527768434

Mishura4 commented:
Changed some mentions of NeoForged to NeoForge and vice versa as requested (see https://github.com/neoforged/NeoForge/pull/1#pullrequestreview-1527768434)

Also updated some docs links that were now dead

Mishura4 committed:
Merge branch '1.20.x' into 1.20.x

DataDalton commented on docs/README.md:
```diff
@@ -29,40 +29,40 @@ For support and questions, visit [the NeoForged Discord server][Discord].
+[Getting-Started]: https://docs.neoforged.net/docs/gettingstarted/
+[NeoForgeDev]: https://docs.neoforged.net/docs/forgedev/
+[Pull-Requests]: https://docs.neoforged.net/docs/forgedev/#making-changes-and-pull-requests
+[CurseForge]: https://curseforge.com/placeholder
```
Should probably remove this weird placeholder link that's broken and just link directly to Curseforge, same for main README

DataDalton commented on docs/README.md:
```diff
@@ -29,40 +29,40 @@ For support and questions, visit [the NeoForged Discord server][Discord].
 
```
Should probably rewrite this original installing NeoForge thing as it doesn't make sense and should say how to install NeoForge without going through Curseforge and linking to the installer on https://neoforged.net/

Technici4n commented:
See #232.
